File: alliances/all-data/webapp/src/types/ecosystem_graph.ts

export interface EcosystemGraphNode {
    id: string,
    name: string,
    description: string,
    type: string
}

export interface EcosystemGraphEdge {
    id: string,
    sourceNodeID: string,
    destinationNodeID: string,
    kind: string
}

export interface EcosystemGraph {
    nodes: EcosystemGraphNode[],
    edges: EcosystemGraphEdge[],
}

export enum LockStatus {
    NotRequested,
    Acquired,
    Busy
}

File: alliances/all-data-provider/model/ecosystem_graph.go
package model

type EcosystemGraphNode struct {
	ID          string `json:"id"`
	Name        string `json:"name"`
	Description string `json:"description"`
	Type        string `json:"type"`
}

type EcosystemGraphEdge struct {
	ID                string `json:"id"`
	SourceNodeID      string `json:"sourceNodeID"`
	DestinationNodeID string `json:"destinationNodeID"`
	Kind              string `json:"kind"`
}

type EcosystemGraphData struct {
	Nodes []*EcosystemGraphNode `json:"nodes"`
	Edges []*EcosystemGraphEdge `json:"edges"`
}

type RefreshLockEcosystemGraphParams struct {
	Nodes     []*EcosystemGraphNode `json:"nodes"`
	Edges     []*EcosystemGraphEdge `json:"edges"`
	UserID    string                `json:"userID"`
	LockDelay int                   `json:"lockDelay"`
}

type DropLockEcosystemGraphParams struct {
	UserID string `json:"userID"`
}
